### PR TITLE
Refactor navigation drawer and support/add various navigation items

### PR DESCRIPTION
**Pull request by ItsPierre**

### CODE REVIEW

**Comment by ItsPierre** [comment]
https://github.com/Crystal-Creations-GbR/crystal-components/compare/5d1f458625f0c79aad12a6b2377ea2935134aa4a..d29ad43ecfa1e0ed3eba5f8d84d73a26dc9e2de7


Made it smaller and the text not bold

File: src/components/NavigationDrawer/CNavigationDrawer.vue
Comment (johannes-huther) [inline src/components/NavigationDrawer/CNavigationDrawer.vue:1]: please use kebab case for the directory

File: src/components/NavigationDrawer/CNavigationSubtitle.vue [hunk 1]
```diff
@@ -42,7 +42,26 @@ const props = withDefaults(
...
 <style lang="scss">
+.c-navigation-subtitle {
+  &.text-subtitle-2 {
+    height: 20px;
+  }
+  &.text-title {
+    height: 25px;
+  }
+}
+
```
Comment (johannes-huther) [inline src/components/NavigationDrawer/CNavigationSubtitle.vue]: this belongs to the first commit, doesn't it?

File: src/components/NavigationDrawer/CNavigationSubtitle.vue [hunk 2]
```diff
@@ -42,7 +42,26 @@ const props = withDefaults(
...
 .v-list .c-navigation-subtitle:not(:first-child) {
   margin-top: 16px;
 }
+
+// When used in navigation drawer, show it on rail mode
```
Comment (johannes-huther) [inline src/components/NavigationDrawer/CNavigationSubtitle.vue]: Do you mean hide?

**Review by johannes-huther — CHANGES_REQUESTED** [review]

File: src/components/NavigationDrawer/CNavigationDrawer.vue
Comment (ItsPierre) [reply src/components/NavigationDrawer/CNavigationDrawer.vue:1]: just took 40 minutes to rebase...

**Comment by ItsPierre** [comment]
https://github.com/Crystal-Creations-GbR/crystal-components/compare/d29ad43ecfa1e0ed3eba5f8d84d73a26dc9e2de7..84b60a417144297bb0d11b356d5a779457503b37

**Review by johannes-huther — APPROVED** [review]

**Pull request merged**